dev/NavigationView/NavigationView.cpp

```diff
@@ -2045,17 +2327,18 @@ bool NavigationView::IsSettingsItem(winrt::IInspectable const& item)
 
 void NavigationView::UnselectPrevItem(winrt::IInspectable const& prevItem, winrt::IInspectable const& nextItem)
 {
-    // ListView already handled unselect by itself if ListView raise SelectChanged by itself.
-    // We only need to handle unselect when:
-    // 1, select from setting to listviewitem or null
-    // 2, select from listviewitem to setting
-    // 3, select from listviewitem to null from API.
     if (prevItem && prevItem != nextItem)
     {
-        if (IsSettingsItem(prevItem) || (nextItem && IsSettingsItem(nextItem)) || !nextItem)
-        {
-            ChangeSelectStatusForItem(prevItem, false /*selected*/);
-        }
+        bool setIgnoreNextSelectionChangeToFalse = !m_shouldIgnoreNextSelectionChange;
+        auto scopeGuard = gsl::finally([this, setIgnoreNextSelectionChangeToFalse]()
```

Comment:
>setIgnoreNextSelectionChangeToFalse [](start = 46, length = 35)

`setIgnoreNextSelectionChangeToFalse = !m_shouldIgnoreNextSelectionChange` and then delete the line above.